Sources/Purchasing/Purchases/PurchaseParams.swift

```diff
@@ -37,12 +45,16 @@ import Foundation
     let winBackOffer: WinBackOffer?
     let metadata: [String: String]?
 
+    /// Options for the confirmIn: parameter of the `purchase(confirmIn:options:)` SK2 APIs.
+    let storeKit2ConfirmInOptions: StoreKit2ConfirmInOptions?
```

Comment:
It got quite complicated to pass the selected UIScene and NSWindow from here all the way down to the purchase() function's call-site since there are so many `#if canImport()`s and `#if os()`s to keep track of. Instead of doing that, I introduced a new internal `StoreKit2ConfirmInOptions` type to make this much easier.